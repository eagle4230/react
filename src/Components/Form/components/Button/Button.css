.button {
    background-color: #fafafa;
    color: blue;
    margin-left: 18px;
    border-radius: 10px;
    border: none;
}