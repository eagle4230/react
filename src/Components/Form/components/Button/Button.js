import './Button.css';

export const Button = ({ label }) => {
  return <button className="button">{label}</button>;
};
