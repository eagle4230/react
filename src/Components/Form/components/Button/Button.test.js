import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { Button } from "./Button";

describe('test_Button', () => {
    it('test_1: рендер кнопки', () => {
        render(<Button label="test" />);
    });

    it('test_2: сравнение с снимком', () => {
        const { asFragment } = render(<Button label="test" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('test_3: поиск текста в рендере', () => {
        render(<Button label="test" />);
        expect(screen.getByText(/test/)).toBeInTheDocument();
    });
})