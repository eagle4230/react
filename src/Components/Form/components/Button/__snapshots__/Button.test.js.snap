// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test_Button test_2: сравнение с снимком 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    test
  </button>
</DocumentFragment>
`;
